app/web_hub.go

```diff
@@ -324,7 +324,10 @@ func (h *Hub) Register(webConn *WebConn) {
 }
 
 func (h *Hub) Unregister(webConn *WebConn) {
-	h.unregister <- webConn
+	select {
+	case h.unregister <- webConn:
+	case <-h.stop:
```

Comment:
WebConns unregister themselves as they close. The Hub goroutine that also processes registration messages closes WebConns. So when WebConns try to unregister themselves from that goroutine, there's a deadlock.

This makes unregister return without doing anything if the hub is stopping.

Reply:
And in case you have trouble remembering the subtleties of channels like I do: closed channels never block.